src/status_im/accounts/access/core.cljs

```diff
@@ -0,0 +1,105 @@
+  (cond (string/blank? recovery-phrase) :required-field
+        (not (mnemonic/valid-phrase? recovery-phrase)) :recovery-phrase-invalid))
+
+(defn check-phrase-warnings [recovery-phrase]
+  (when (not (mnemonic/status-generated-phrase? recovery-phrase))
+    :recovery-phrase-unknown-words))
+
+(fx/defn set-phrase [{:keys [db]} masked-recovery-phrase]
+  (let [recovery-phrase (security/unmask masked-recovery-phrase)]
+    {:db (update db :accounts/access assoc :passphrase (string/lower-case recovery-phrase))}))
+
+(fx/defn validate-phrase [{:keys [db]}]
+  (let [recovery-phrase (get-in db [:accounts/access :passphrase])]
+    {:db (update db :accounts/access assoc
+                 :passphrase-error (check-phrase-errors recovery-phrase)
+                 :passphrase-warning (check-phrase-warnings recovery-phrase))}))
+
+(defn access-account! [masked-passphrase password]
+  (status/access-account
+   (mnemonic/sanitize-passphrase (security/unmask masked-passphrase))
+   password
+   (fn [result]
+     ;; here we deserialize result, dissoc mnemonic and serialize the result again
+     ;; because we want to have information about the result printed in logs, but
+     ;; don't want secure data to be printed
+     (let [data (-> (types/json->clj result)
+                    (dissoc :mnemonic)
+                    (types/clj->json))]
+       (re-frame/dispatch [:accounts.access.callback/access-account-success data password])))))
+
+(fx/defn validate-access-result [{:keys [db] :as cofx} {:keys [error pubkey address]} password]
+  (if (empty? error)
+    (let [account {:pubkey     pubkey
+                   :address    address
+                   :photo-path (identicon/identicon pubkey)
+                   :mnemonic   ""}]
+      (accounts.create/on-account-created cofx account password true))
+    {:db (assoc-in db [:accounts/access :error] (i18n/label :t/recover-password-invalid))}))
+
+(fx/defn on-account-accessed [cofx result password]
+  (let [data (types/json->clj result)]
+    (fx/merge cofx
+              (validate-access-result data password))))
+
+(defn access-account [{:keys [db]}]
+  (let [{:keys [password passphrase]} (:accounts/access db)]
+    {:db (assoc-in db [:accounts/access :processing?] true)
+     :accounts.access/access-account [(security/mask-data passphrase) password]}))
+
+(defn access-account-with-checks [{:keys [db] :as cofx}]
+  (let [{:keys [passphrase processing?]} (:accounts/access db)]
+    (when-not processing?
+      (if (mnemonic/status-generated-phrase? passphrase)
+        (access-account cofx)
+        {:ui/show-confirmation
+         {:title               (i18n/label :recovery-typo-dialog-title)
+          :content             (i18n/label :recovery-typo-dialog-description)
+          :confirm-button-text (i18n/label :recovery-confirm-phrase)
+          :on-accept           #(re-frame/dispatch [:accounts.access.ui/access-account-confirmed])}}))))
+
+(fx/defn next-step [{:keys [db] :as cofx} step]
```

Comment:
the logic of accounts.create is pretty much the same, did you try to factorize ?

Reply:
Not yet, will be done later.
Basically, the answer is the same as the previous one — there is absolutely no need to do it here in this PR. There are a lot of things that can be improved on these screens in general — we need to make everything more generic, use parts of the code, but when you start diving into it, you understand that it's huge and it's not something that needs to be done for a story marked as "critical". 
However, I will start doing it after I finish with current changes.